src/transformStateWithClones.js

```
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      currentState = {};
+    } else if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      currentState = { ...currentState };
+
+      for (const key of action.keysToRemove) {
+        delete currentState[key];
+      }
+    }
```

There is no default/error-handling branch for unknown action types. This violates checklist item #3: "[CODE STYLE]: switch/case should always have default case for error handling." When you convert to a switch add a `default` that either throws a descriptive Error or otherwise handles unknown action types in a controlled way (do not let unknown types be silently ignored) .